### PR TITLE
Refactor TxFactoryBuilder

### PR DESCRIPTION
**Pull request by marco-scoppetta**

### CODE REVIEW

File: grakn-engine/src/test/java/ai/grakn/engine/controller/GraqlControllerTest.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public void whenRunningGetQuery_EnsureAllInstancesAreReturned() {
         assertEquals(expectedInstances, instances);
     }
 
-    //TODO: THis test should be improved
+    //TODO: This test should be improved
```
Comment (marco-scoppetta) [inline grakn-engine/src/test/java/ai/grakn/engine/controller/GraqlControllerTest.java]: Improved! 👍

File: grakn-kb/src/main/java/ai/grakn/factory/TxFactoryAbstract.java [hunk 2]
```diff
@@ -25,99 +25,113 @@
...
 
 import javax.annotation.CheckReturnValue;
+import javax.annotation.Nullable;
 
 import static javax.annotation.meta.When.NEVER;
 
 /**
  * <p>
- *     Defines the construction of Grakn Graphs
+ * Defines the construction of {@link GraknTx} based on
```
Comment (aelred) [inline grakn-kb/src/main/java/ai/grakn/factory/TxFactoryAbstract.java]: based on what? 🤔

File: grakn-kb/src/main/java/ai/grakn/factory/EmbeddedGraknSession.java [hunk 3]
```diff
@@ -129,27 +137,19 @@ public CommitLogHandler commitLogHandler(){
      * A user should not call this method directly.
      * See {@link Grakn#session(String, String)} for creating a {@link GraknSession} for the remote API
      */
-    @SuppressWarnings("unused")//This must remain public because it is accessed via reflection
-    public static EmbeddedGraknSession create(Keyspace keyspace, SimpleURI engineUri){
-        return create(keyspace, engineUri.toString());
-    }
```
Comment (aelred) [inline grakn-kb/src/main/java/ai/grakn/factory/EmbeddedGraknSession.java]: We should keep this method, since it takes a `SimpleURI`, not a `String`

Comment (marco-scoppetta) [reply grakn-kb/src/main/java/ai/grakn/factory/EmbeddedGraknSession.java]: It's unused for now, I think we should have just one method and  in the future this method should accept `SimpleURI` - but for that we should do a full refactor (given that `EmbeddedGraknSession` uses `uri` as a String )- for now having two methods was just confusing to me 🤔 🥇

**Comment by marco-scoppetta** [comment]
!rtg

**Pull request merged**